diff -crB tmp/ghc-7.0.1.20110217//configure.ac ghc-7.0.1.20110217//configure.ac
*** tmp/ghc-7.0.1.20110217//configure.ac	2011-02-19 20:10:05.000000000 +0200
--- ghc-7.0.1.20110217//configure.ac	2011-02-22 10:53:08.020609496 +0200
***************
*** 34,40 ****
  AC_SUBST([CONFIGURE_ARGS], [$ac_configure_args])
  
  dnl ----------------------------------------------------------
! dnl ** Find unixy sort and find commands, 
  dnl ** which are needed by FP_SETUP_PROJECT_VERSION
  
  dnl ** Find find command (for Win32's benefit)
--- 34,40 ----
  AC_SUBST([CONFIGURE_ARGS], [$ac_configure_args])
  
  dnl ----------------------------------------------------------
! dnl ** Find unixy sort and find commands,
  dnl ** which are needed by FP_SETUP_PROJECT_VERSION
  
  dnl ** Find find command (for Win32's benefit)
***************
*** 255,260 ****
--- 255,269 ----
      esac
  }
  
+ BuildVendor=`echo "$BuildVendor" | sed -e 's/pc/unknown/'`
+ [BuildArch=`echo "$BuildArch" | sed -e 's/i[4-6]86/i386/'`]
+ 
+ HostVendor=`echo "$HostVendor" | sed -e 's/pc/unknown/'`
+ [HostArch=`echo "$HostArch" | sed -e 's/i[4-6]86/i386/'`]
+ 
+ TargetVendor=`echo "$TargetVendor" | sed -e 's/pc/unknown/'`
+ [TargetArch=`echo "$TargetArch" | sed -e 's/i[4-6]86/i386/'`]
+ 
  BuildPlatform="$BuildArch-$BuildVendor-$BuildOS"
  BuildPlatform_CPP=`echo "$BuildPlatform" | sed -e 's/\./_/g' -e 's/-/_/g'`
  BuildArch_CPP=`    echo "$BuildArch"     | sed -e 's/\./_/g' -e 's/-/_/g'`
***************
*** 287,299 ****
  
  # Verify that the installed (bootstrap) GHC is capable of generating
  # code for the requested build platform.
! if test "$build" != "$bootstrap_target"
! then
!     echo "This GHC (${WithGhc}) does not generate code for the build platform"
!     echo "   GHC target platform    : $bootstrap_target"
!     echo "   Desired build platform : $BuildPlatform"
!     exit 1
! fi
  
  AC_SUBST(BuildPlatform)
  AC_SUBST(HostPlatform)
--- 296,308 ----
  
  # Verify that the installed (bootstrap) GHC is capable of generating
  # code for the requested build platform.
! # if test "$build" != "$bootstrap_target"
! # then
! #     echo "This GHC (${WithGhc}) does not generate code for the build platform"
! #     echo "   GHC target platform    : $bootstrap_target"
! #     echo "   Desired build platform : $BuildPlatform"
! #     exit 1
! # fi
  
  AC_SUBST(BuildPlatform)
  AC_SUBST(HostPlatform)
